meos/src/point/tpoint_restrfuncs.c

```diff
@@ -0,0 +1,2584 @@
+    NORMALIZE_NO);
+
+  if (atfunc)
+    return result_at;
+
+  /* If "minus" compute the complement wrt time */
+  SpanSet *ps1 = tsequence_time(seq);
+  SpanSet *ps2 = tsequenceset_time(result_at);
+  SpanSet *ps = minus_spanset_spanset(ps1, ps2);
+  TSequenceSet *result_comp = NULL;
+  if (ps)
+  {
+    result_comp = tcontseq_restrict_periodset(seq, ps, REST_AT);
+    pfree(ps);
+  }
+  pfree(ps1); pfree(ps2); pfree(result_at);
+  return result_comp;
+}
+
+/*****************************************************************************/
+
+/**
+ * @brief Restrict the temporal point to the spatial dimensions of a
+ * spatiotemporal box
+ * @param[in] seq Temporal point sequence
+ * @param[in] box Bounding box
+ * @param[in] border Do we need to remove the max border?
+ * @param[out] count Number of elements in the resulting array
+ * @pre The sequence is simple in order to recover the time dimension from
+ * the result of the Cohen-Sutherland line clipping algorithm
+ */
+TSequence **
+tpointseq_linear_at_stbox_xyz(const TSequence *seq, const STBox *box,
+  bool border, int *count)
+{
+  assert(MEOS_FLAGS_GET_INTERP(seq->flags) == LINEAR);
+  TSequence **result = NULL;
+  *count = 0;
+
+  /* Instantaneous sequence */
+  if (seq->count == 1)
+  {
+    const TInstant *inst = TSEQUENCE_INST_N(seq, 0);
+    if (tpointinst_restrict_stbox_iter(inst, box, border, REST_AT))
+    {
+      result = palloc(sizeof(TSequence *));
+      result[0] = tsequence_copy(seq);
+      *count = 1;
+    }
+    return result;
+  }
+
+  /* General case */
+  bool hasz_seq = MEOS_FLAGS_GET_Z(seq->flags);
+  bool hasz_box = MEOS_FLAGS_GET_Z(box->flags);
+  bool hasz = hasz_seq && hasz_box;
+  result = palloc(sizeof(TSequence *) * seq->count);
+  const TInstant *inst1 = TSEQUENCE_INST_N(seq, 0);
+  Datum p1 = tinstant_value(inst1);
+  bool lower_inc = seq->period.lower_inc;
+  int k = 0;
+  for (int i = 1; i < seq->count; i++)
+  {
+    const TInstant *inst2 = TSEQUENCE_INST_N(seq, i);
+    bool upper_inc = (i == seq->count - 1) ? seq->period.upper_inc : false;
+    Datum p2 = tinstant_value(inst2);
+    Datum p3, p4;
+    bool found;
+    TInstant *instants[2];
+    if (datum2_point_eq(p1, p2))
+    {
+      /* Constant segment */
+      if (hasz)
+      {
+        const POINT3DZ *pt1 = DATUM_POINT3DZ_P(p1);
+        found = (box->xmin <= pt1->x && pt1->x <= box->xmax) &&
+                (box->ymin <= pt1->y && pt1->y <= box->ymax) &&
+                (box->zmin <= pt1->z && pt1->z <= box->zmax);
+      }
+      else
+      {
+        const POINT2D *pt1 = DATUM_POINT2D_P(p1);
+        found = (box->xmin <= pt1->x && pt1->x <= box->xmax) &&
+                (box->ymin <= pt1->y && pt1->y <= box->xmax);
+      }
+      if (found)
+      {
+        instants[0] = (TInstant *) inst1;
+        instants[1] = (TInstant *) inst2;
+        result[k++] = tsequence_make((const TInstant **) instants, 2,
+          lower_inc, upper_inc, LINEAR, NORMALIZE_NO);
+      }
+    }
```

Comment:
Not sure this handles the lower_inc/upper_inc correctly.
Should probably use the same technique as in `tpointinst_restrict_stbox_iter` here.